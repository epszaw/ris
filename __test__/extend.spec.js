const { schema, extend, validate } = require('src/index')

describe('extend', () => {
  it('should add new rules to schema and allows to use it', () => {
    extend('even', v => v % 2 === 0, 'Even')

    const extendesSchema = {
      foo: schema.even,
    }

    expect(
      validate(extendesSchema, {
        foo: 2,
      }),
    ).toEqual({
      foo: 2,
    })
    expect(() => {
      validate(extendesSchema, {
        foo: 3,
      })
    }).toThrowErrorMatchingSnapshot()
  })

  it('should add new rules to schema with custom error message and allows to use it', () => {
    extend(
      'even',
      v => v % 2 === 0,
      v => `Given value must be even number, got ${v}`,
    )

    const extendesSchema = {
      foo: schema.even,
    }

    expect(() => {
      validate(extendesSchema, {
        foo: 3,
      })
    }).toThrowErrorMatchingSnapshot()
  })
})
