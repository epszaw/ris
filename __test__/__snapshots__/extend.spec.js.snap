// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extend should add new rules to schema and allows to use it 1`] = `"Validation error: foo expected Even got 3"`;

exports[`extend should add new rules to schema with custom error message and allows to use it 1`] = `"Given value must be even number, got 3"`;
